src/network/networkd-wwan.c

```diff
@@ -0,0 +1,437 @@
+        Address *address;
+        Route *route;
+        int r, ret = 0;
+
+        assert(link);
+
+        if (!IN_SET(link->state, LINK_STATE_CONFIGURING, LINK_STATE_CONFIGURED))
+                return 0;
+
+        /* First, mark bearer configs. */
+        SET_FOREACH(address, link->addresses) {
+                if (address->source != NETWORK_CONFIG_SOURCE_MODEM_MANAGER)
+                        continue;
+
+                address_mark(address);
+        }
+
+        SET_FOREACH(route, link->routes) {
+                if (route->source != NETWORK_CONFIG_SOURCE_MODEM_MANAGER)
+                        continue;
+
+                route_mark(route);
+        }
+
+        if (b && FLAGS_SET(b->ip_type, ADDRESS_FAMILY_IPV4)) {
+                if (b->connected && b->ip4_method == MM_BEARER_IP_METHOD_STATIC) {
+                        r = link_request_bearer_address(link, AF_INET, &b->ip4_address, b->ip4_prefixlen);
+                        if (r < 0)
+                                return r;
+
+                        r = link_request_bearer_route(link, AF_INET, &b->ip4_gateway, &b->ip4_address);
+                        if (r < 0)
+                                return r;
+                }
+
+                if (b->connected && b->ip4_method == MM_BEARER_IP_METHOD_DHCP) {
+                        if (!link_dhcp4_enabled(link))
+                                log_link_notice(link, "The WWAN connection requested DHCPv4 client, but it is disabled.");
+
+                        r = dhcp4_start(link);
+                        if (r < 0)
+                                return log_link_warning_errno(link, r, "Failed to start DHCPv4 client: %m");
+                } else {
+                        r = sd_dhcp_client_stop(link->dhcp_client);
+                        if (r < 0)
+                                ret = log_link_warning_errno(link, r, "Could not stop DHCPv4 client: %m");
+                }
+        }
+
+        if (b && FLAGS_SET(b->ip_type, ADDRESS_FAMILY_IPV6)) {
+                if (b->connected && b->ip6_method == MM_BEARER_IP_METHOD_STATIC) {
+                        r = link_request_bearer_address(link, AF_INET6, &b->ip6_address, b->ip6_prefixlen);
+                        if (r < 0)
+                                return r;
+
+                        r = link_request_bearer_route(link, AF_INET6, &b->ip6_gateway, NULL);
+                        if (r < 0)
+                                return r;
+                }
+
+                if (b->connected && b->ip6_method == MM_BEARER_IP_METHOD_DHCP) {
+                        if (!link_ipv6_accept_ra_enabled(link) && !link_dhcp6_enabled(link))
+                                log_link_notice(link,
+                                                "The WWAN connection requested IPv6 dynamic address configuration,"
+                                                "but both IPv6 Router Discovery and DHCPv6 client are disabled.");
```

Comment:
Same here.

Reply:
Also here, `ndisc_start()` and `dhcp6_start()` do nothing if they are disabled.